### PR TITLE
feat(bun): Add support for updating text-format lockfile when package.json changes

### PR DESCRIPTION
**Pull request by nathankleyn**
## Changes

Bun 1.2.x will introduce and switch by default to a text-format verison of the lockfile. Where currently they use a binary version named `bun.lockb`, this will become `bun.lock` and be plaintext:

https://bun.sh/blog/bun-lock-text-lockfile

This can be switched on by default in-advance since Bun 1.1.40 and so will appear in the wild as of now:

https://bun.sh/blog/bun-v1.1.40#new-install-savetextlockfile-in-bunfig-toml

This commit adds support for both lockfile formats by supporting by file-suffixes.

## Context

As of Bun 1.1.40, this feature can be opted-in and Renovate no longer works for Bun without it.

As of Bun 1.2.x this feature will be on by default.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

The docs that would be affected are auto-generated ones

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Tested and run on https://github.com/nathankleyn/bun-renovate-text-lockfile-test, where you can see a PR at which correctly detected and changed the new-style lockfile: https://github.com/nathankleyn/bun-renovate-text-lockfile-test/pull/2

### CODE REVIEW

File: lib/modules/manager/bun/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const supersedesManagers = ['npm'];
 export const supportsLockFileMaintenance = true;
 
 export const defaultConfig = {
-  fileMatch: ['(^|/)bun\\.lockb$'],
+  fileMatch: ['(^|/)bun\\.lockb$', '(^|/)bun\\.lock$'],
```
Comment (rarkins) [inline lib/modules/manager/bun/index.ts:16]:
```suggestion
  fileMatch: ['(^|/)bun\\.lockb?$'],
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs testing on a real repo with text lockfile format

**Comment by nathankleyn** [comment]
@rarkins Thanks so much for super-fast review — I've made that fix to simplify the regex, and I've created a public repository where I've run Renovate with the changes at https://github.com/nathankleyn/bun-renovate-text-lockfile-test — you can see a PR at which correctly detected and changed the new-style lockfile: https://github.com/nathankleyn/bun-renovate-text-lockfile-test/pull/2

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
@nathankleyn thanks for your PR! I'm surprised/delighted to discover that this started working with very little logic change.

**Review by rarkins — APPROVED** [review]

**Comment by glensc** [comment]
`verison` typo in pr body

**Pull request merged**